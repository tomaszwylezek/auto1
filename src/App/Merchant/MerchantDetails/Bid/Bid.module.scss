.bid {
  max-width: 300px;
  font-size: 0.6em;
}
