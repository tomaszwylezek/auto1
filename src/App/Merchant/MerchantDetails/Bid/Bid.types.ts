import { IBid } from 'model';

export interface IBidOwnProps extends IBid {}

export interface IBidProps extends IBidOwnProps {}
