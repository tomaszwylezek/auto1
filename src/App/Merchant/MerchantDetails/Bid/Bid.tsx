import React, { FC } from 'react';
import classNames from 'classnames';

import styles from './Bid.module.scss';

// Models
import { IBidProps } from 'App/Merchant/MerchantDetails/Bid/Bid.types';

export const Bid: FC<IBidProps> = ({ id, carTitle, amount, created }) => {
  return (
    <div className={classNames('card text-white bg-secondary mb-3 mr-2 ', styles.bid)}>
      <div className="card-header py-1 px-3">{id}</div>
      <div className="card-body p-3">
        <h5 className="card-title">
          {carTitle} - {amount}
          $$
        </h5>
        <p className="card-text">Created: {created}</p>
      </div>
    </div>
  );
};
