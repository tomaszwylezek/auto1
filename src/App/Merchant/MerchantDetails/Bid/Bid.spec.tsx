import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';

import { Bid } from './Bid';

// Models
import { IBidProps } from './Bid.types';

describe('Bid component should', (): void => {
  let props: IBidProps;
  let component: ShallowWrapper<IBidProps>;

  beforeEach((): void => {
    props = {
      id: '3',
      amount: 12,
      carTitle: 'Toyota',
      created: new Date('Sat Jul 06 2019 10:31:54 GMT+0200 (Central European Summer Time)').toString()
    };

    component = shallow(<Bid {...props} />);
  });

  it('render correctly', (): void => {
    expect(component).toMatchSnapshot();
  });
});
