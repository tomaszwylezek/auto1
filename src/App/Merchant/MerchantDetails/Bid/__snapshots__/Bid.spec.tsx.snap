// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bid component should render correctly 1`] = `
<div
  className="card text-white bg-secondary mb-3 mr-2  bid"
>
  <div
    className="card-header py-1 px-3"
  >
    3
  </div>
  <div
    className="card-body p-3"
  >
    <h5
      className="card-title"
    >
      Toyota
       - 
      12
      $$
    </h5>
    <p
      className="card-text"
    >
      Created: 
      Sat Jul 06 2019 10:31:54 GMT+0200 (Central European Summer Time)
    </p>
  </div>
</div>
`;
